### ISSUE TITLE
desi_daily_proc_manager add tilenight support

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1908]
Add support for tilenight to `desi_daily_proc_manager` (currently it is only supported by `desi_run_night` via `--use-tilenight` opt-in).  This will be important for efficient Perlmutter daily processing.

Consider switching `--use-tilenight` to be a boolean option:
* if not specified (None), use False on Cori and True on Perlmutter
* otherwise require True or False to opt in/out of the default on that system

i.e. I keep forgetting to specify `--use-tilenight` on Perlmutter, and only realize it when it submits 3x more jobs than I intended.

**Comment by sbailey** [comment]
One key thing that desi_daily_proc_manager does that desi_run_night does not is the creation of the exposure tables: desi_run_night requires them to pre-exist, while desi_daily_proc_manager creates them on-the-fly.  In the specific current case of sneakernet with all data arriving at once, we can circumvent this with desi_create_exposure_tables + desi_run_night (I think; I've personally never done that), but I think desi_daily_proc_manager is the only thing that can monitor for new data, update the exposure tables on the fly, and submit jobs as needed.

**Comment by marcelo-alvarez** [comment]
@sbailey thanks for pointing out this crucial difference. 

My understanding there are two cases we need for Perlmutter:
1. data for a night is transferred all at once, as it is now
2. data is transferred in real time throughout a night

For (1) we should be able to run a new night either (a) by using desi_create_exposure_tables + desi_run_night or (b) by adding --use-tilenight to desi_daily_proc_manager. 

For (2) adding --use-tilenight to desi_daily_proc_manager does not work on Perlmutter, since there is no way to leave it running indefinitely. A scrontab job that wakes up periodically to update exposure tables and run a modified version of desi_run_night seems the simplest approach going forwared, but this could be beyond the slope of this issue.

Returning to (1) it seems preferable to use option (1a), since option (1b) would take more time and may not be useful for (2).

**Comment by sbailey** [comment]
I took another look at desi_run_night -> desispec.scripts.submit_night, and agree that it isn't a trivial update to add the same logic to desi_daily_proc_manager.  Followup comments:

With perlmutter scron, the "cron" qos allows jobs to run up to 24 hours, and the "workflow" qos allows up to 90 days, so our original expectation that we had to update desi_daily_proc_manager to be relaunched frequently while saving state in between appears to have been wrong (or perhaps they updated the policy in response to needs like ours).  i.e. it looks like we don't have to do that part of the refactor, and can keep the model of launching desi_daily_proc_manager once a day and letting it run until morning.

We should treat Perlmutter as the primary machine and get desi_daily_proc_manager working well with that (including using tilenight), and let Cori be the special case.  e.g. plumbing tilenight into desi_daily_proc_manager might be easier if we *don't* try to launch individual exposures prior to getting all exposures for a tile.  i.e. follow the tilenight logic of waiting for all exposures on a tile, and then submit either a single tilenight script, or prestdstar*N + stdstar + poststdstar*N jobs, or maybe just stick with tilenight on Cori too and accept that it takes a long time to run.
